openmdao/docs/usr-guide/tutorials/implicit.rst

```diff
@@ -0,0 +1,251 @@
+
+                x = params['x']
+                a = self.a
+                b = self.b
+                J = {}
+
+                J['y', 'x'] = 2.0*a*x + b
+                return J
+
+We have made these components slightly more general so that you can, for
+example, change the slope and intercept of the `Line` to try solving
+different problems.
+
+Now we need to add a component that defines a residual for the difference
+between "parabola.y" and "line.y". We want to let an OpenMDAO solver drive
+this difference to zero.
+
+.. testcode:: Solver1
+
+        class Balance(Component):
+            """Evaluates the residual y1-y2"""
+
+            def __init__(self):
+                super(Balance, self).__init__()
+
+                self.add_param('y1', 0.0)
+                self.add_param('y2', 0.0)
+                self.add_state('x', 5.0)
+
+            def solve_nonlinear(self, params, unknowns, resids):
+                """This component does no calculation on its own. It mainly holds the
+                initial value of the state. An OpenMDAO solver outside of this
+                component varies it to drive the residual to zero."""
+                pass
+
+            def apply_nonlinear(self, params, unknowns, resids):
+                """ Report the residual y1-y2 """
+
+                y1 = params['y1']
+                y2 = params['y2']
+
+                resids['x'] = y1 - y2
+
+            def linearize(self, params, unknowns, resids):
+                """ Jacobian for our parabola."""
+
+                J = {}
+                J['x', 'y1'] = 1.0
+                J['x', 'y2'] = -1.0
+                return J
+
+This component holds both our state and the residual. This component produces
+no explicit outputs, so the `solve_nonlinear` method doesn't do anything (but
+it still must be declared). In the `apply_nonlinear` method, we take the
+difference "y1-y2" and place it in the residual for "x". The derivatives are
+straightforward.
+
+Note that the residual equation is not a direct function of the state, but it
+is indirectly a function via y1 and y2. The partial derivative of the
+residual with respect to 'x' is zero, though the total derivative calculated
+by OpenMDAO of the residual with respect to 'x' is nonzero.
+
+Finally, lets set up the model.
+
+.. testcode:: Solver1
+
+    top = Problem()
+    root = top.root = Group()
+    root.add('line', Line())
+    root.add('parabola', Parabola())
+    root.add('bal', Balance())
+
+    root.connect('line.y', 'bal.y1')
+    root.connect('parabola.y', 'bal.y2')
+    root.connect('bal.x', 'line.x')
+    root.connect('bal.x', 'parabola.x')
+
+    root.nl_solver = Newton()
+    root.ln_solver = ScipyGMRES()
+
+    top.setup()
+
+Here we connect the output of the Line and Parabola component to the params
+of the Balance component. The state on "Balance" feeds the params on both
+components.
+
+To solve this system, we need to slot a specify a solver in "root.nl_solver".
```

Comment:
Don't use the word "slot", since it relates to classic. Also, typo here. Suggest chaning to "to solve this system we need to specify a solver"